rotkehlchen/db/dbhandler.py

```diff
@@ -631,35 +631,40 @@ def set_settings(self, write_cursor: 'DBCursor', settings: ModifiableDBSettings)
         CachedSettings().update_entries(settings)
 
     @overload
-    def get_cache(self, cursor: 'DBCursor', name: DBCache) -> Timestamp:
+    def get_cache(self, cursor: 'DBCursor', name: DBCache | str) -> Timestamp | None:
         ...
 
     @overload
-    def get_cache(self, cursor: 'DBCursor') -> dict[DBCache, Timestamp]:
+    def get_cache(self, cursor: 'DBCursor') -> dict[DBCache | str, Timestamp]:
         ...
 
     def get_cache(
             self,
             cursor: 'DBCursor',
-            name: DBCache | None = None,
-    ) -> dict[DBCache, Timestamp] | Timestamp:
+            name: DBCache | str | None = None,
+    ) -> dict[DBCache | str, Timestamp] | Timestamp | None:
         """Returns the cache values from the `key_value_cache` table of the DB
         - If `name` is not passed, it returns all the values in the table
-        - If `name` is passed, it returns the value of the given name, defaults to Timestamp(0)"""
+        - If `name` is passed, it returns the value of the given name, defaults to None"""
         if name is None:
             cursor.execute('SELECT name, value FROM key_value_cache;')
             return {
                 DBCache.deserialize(q[0]): Timestamp(int(q[1]))
                 for q in cursor
             }
-        value = cursor.execute('SELECT value FROM key_value_cache WHERE name=?;', (name.value,)).fetchone()  # noqa: E501
-        return Timestamp(0) if value is None else Timestamp(int(value[0]))
+        value = cursor.execute(
+            'SELECT value FROM key_value_cache WHERE name=?;',
+            (name.value if isinstance(name, DBCache) else name,),
+        ).fetchone()
```

Comment:
instead of doing this wouldn't be better to filter only by the ones in `DBCache`? When do you need them all? I see that you query them directly from the table instead of using the get_cache/set_cache logic